code/modules/clothing/outfits/factions/independent.dm

```diff
@@ -30,8 +30,14 @@
 	..()
 	if(visualsOnly)
 		return
-	var/obj/item/card/id/W = H.wear_id
-	W.access += list(ACCESS_KITCHEN)
+	if(istype(H.wear_id, /obj/item/storage/wallet))
+		var/obj/item/storage/wallet/W = H.wear_id
+		if(W.front_id)
+			var/obj/item/card/id/ID = W.GetID()
+			ID.access += list(ACCESS_KITCHEN)
+	else
+		var/obj/item/card/id/W = H.wear_id
+		W.access += list(ACCESS_KITCHEN)
```

Comment:
```suggestion
	var/obj/item/card/id/W = H.get_idcard()
	W.access += list(ACCESS_KITCHEN)
```

Reply:
thank you oh someone who knows what they are doing looking at this